WordPress/src/main/java/org/wordpress/android/ui/ShareIntentReceiverActivity.java

```diff
@@ -4,233 +4,189 @@
-            mBlogSpinner.setAdapter(adapter);
-            mBlogSpinner.setOnItemSelectedListener(this);
+            loadState(savedInstanceState);
         }
+    }
 
-        loadLastUsed();
-
-        // If type is text/plain hide Media Gallery option
-        if (isSharingText()) {
-            mActionIndex = ADD_TO_NEW_POST;
-            mActionGroup.setVisibility(View.GONE);
-            findViewById(R.id.action_spinner_title).setVisibility(View.GONE);
-            // if text/plain and only one blog, then don't show this fragment, share it directly to a new post
-            if (mSiteNames.length == 1) {
-                // Single site, startActivityAndFinish will pick the first one by default
-                startActivityAndFinish(new Intent(this, EditPostActivity.class));
+    @Override
+    protected void onResume() {
+        super.onResume();
+        // keep setBackground in the onResume, otherwise the transition between activities is visible to the user when
+        // sharing text with an account with one visible site
+        findViewById(R.id.main_view).setBackgroundResource(R.color.login_background_color);
+    }
+
+    protected void refreshContent(){
+        if (FluxCUtils.isSignedInWPComOrHasWPOrgSite(mAccountStore, mSiteStore)) {
+            List<SiteModel> visibleSites = mSiteStore.getVisibleSites();
+            if (visibleSites.size() == 0) {
+                ToastUtils.showToast(this, R.string.cant_share_no_visible_blog, ToastUtils.Duration.LONG);
+                finish();
+            } else if (visibleSites.size() == 1 && isSharingText()) {
+                // if text/plain and only one blog, then don't show the fragment, share it directly to a new post
+                share(ShareAction.SHARE_TO_POST, visibleSites.get(0).getId());
+            }else {
+                // display a fragment with list of sites and list of actions the user can perform
+                initShareFragment(false);
             }
         } else {
-            mActionGroup.setOnCheckedChangeListener(new RadioGroup.OnCheckedChangeListener() {
-                @Override
-                public void onCheckedChanged(RadioGroup group, @IdRes int checkedId) {
-                    mActionIndex = getActionIndex(checkedId);
-                }
-            });
+            // start the login flow and wait onActivityResult
+            ActivityLauncher.loginForShareIntent(this);
         }
     }
 
-    @Override
-    public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
-        if (parent.getId() == R.id.blog_spinner) {
-            mSelectedSiteLocalId = mSiteIds[position];
-        }
+    protected void initShareFragment(boolean afterLogin) {
+        ShareIntentReceiverFragment shareIntentReceiverFragment = ShareIntentReceiverFragment
+            .newInstance(!isSharingText(), loadLastUsedBlogLocalId(), afterLogin);
```

Comment:
I was using googleJavaStyle instead of AospJavaStyle. My bad. Thanks! 

No reason to apologize :). Tbh, I'm usually the one who is being pedantic about code formatting. I like to have a codestyle.xml included in the repository, so everyone has the same settings.